### PR TITLE
Backup the Cloud SQL database to GCS every 6 hours

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes https://github.com/google/exposure-notifications-verification-server/issues/864

**Nuances**

- Cloud SQL doesn't seem to support per-instance IAM, so we have to grant at the project level (yuck, but it's only viewer)

- I'm relying on bucket versioning instead of actually naming the exports with a datetime. This is mostly a design decision to avoid needing a middle service. Ideally Cloud Scheduler would support a template syntax, but it does not.

- The backups aren't compressed (for the same reason as above), but our last backup was 2MB. I'm not really worried about large databases for this project.

- Backup time is configurable via the cron notation.

**Release Note**

```release-note
Automatically backup the Cloud SQL database to GCS every 6 hours
```

/assign @jeremyfaller @icco

### CODE REVIEW

**Review by mikehelmick — CHANGES_REQUESTED** [review]
suggestion - we add GCS retention timeline on the bucket as well

**Review by mikehelmick — APPROVED** [review]

File: terraform/database.tf [hunk 1]
```diff
@@ -285,6 +285,95 @@ resource "null_resource" "migrate" {
   ]
 }
 
+# Create a storage bucket where database backups will be housed.
+resource "google_storage_bucket" "backups" {
+  project  = var.project
+  name     = "${var.project}-backups"
+  location = var.storage_location
+
+  force_destroy               = true
```
Comment (jeremyfaller) [inline terraform/database.tf:294]: feels safer to me to have this be false(the default).

**Review by icco — APPROVED** [review]
/lgtm

**Review by jeremyfaller — COMMENTED** [review]
lgtm, I'll let @icco finish here.

File: terraform/database.tf (hunk 1 above)
Comment (sethvargo) [reply terraform/database.tf:294]: This tells terraform to delete the objects in the bucket before deleting the bucket. GCS doesn't let you delete a bucket with objects inside.

Comment (jeremyfaller) [reply terraform/database.tf:294]: Yeah, I understood that. I felt it was kinda safer to require the bucket be empty.

**Pull request merged**